include/seastar/kafka/producer/kafka_producer.hh

```diff
@@ -41,7 +41,7 @@ private:
 
     std::string _client_id;
     lw_shared_ptr<connection_manager> _connection_manager;
-    partitioner _partitioner;
+    partitioner* _partitioner = new basic_partitioner();
```

Comment:
Move the initialization to `kafka_producer` constructor.

Currently it causes memory leak (`new` without `delete`) - maybe use `unique_ptr` instead?